### ISSUE TITLE
Consider adding a respective stop() method for NodeEngagementMutex

### ISSUE THREAD

**Issue by derekpierre** [nucypher/nucypher#2189]
Follow-up PR related to #2177 

See comment: https://github.com/nucypher/nucypher/pull/2177/files#r470816587

Basically, we could possible provide a `stop()` method for the NodeEngagementMutex to forcibly stop the mutex to unblock and cleanup the threadpool - could be especially useful in failed tests.

**Comment by derekpierre** [comment]
This has gone stale. Mutex is no longer used in the same way as before.
